clayui.com/src/components/IconSearch/index.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * © 2020 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import React, {useEffect, useState} from 'react';
+
+const spritemap = '/images/icons/icons.svg';
+
+const IconSearch = source => {
+	const [searchQuery, setSearchQuery] = useState('');
+	const [filteredIconNames, setFilteredIconNames] = useState([]);
+	const list = filteredIconNames.length ? filteredIconNames : [];
+
+	source = [...source.source];
```

Comment:
Cloning the array here seems pretty odd... When we clone items for the purpose of mutation, I think we should avoid doing is on a large scale here(being in the main body of the component function), and instead we should only mutate for small intermediate steps like the example below.

```js
const pushItem = (items, newItem) => {
     const clonedItems = {...items};

     clonedItems.total++
     clonedItems.items = [...clonedItems.items, newItem];

     return clonedItems;
};
```

Reply:
This and the destructuring from above is related. Destructuring when declaring the param crossed my mind but I disregarded it before even trying, idk why, must have spaced out.

The cloning here was just to get the `source` to be an array instead of an object, which I'll do via destructuring